frontend/web/components/pages/ProjectSettingsPage.js

```diff
@@ -498,6 +603,315 @@ const ProjectSettingsPage = class extends Component {
+                              header={
+                                <Row className='table-header'>
+                                  <Flex className='table-column px-3'>
+                                    Name
+                                  </Flex>
+                                  <div
+                                    className='table-column'
+                                    style={{ width: metadataWidth[0] }}
+                                  >
+                                    Environment
+                                  </div>
+                                  <div
+                                    className='table-column'
+                                    style={{ width: metadataWidth[1] }}
+                                  >
+                                    Segment
+                                  </div>
+                                  <div
+                                    className='table-column'
+                                    style={{ width: metadataWidth[2] }}
+                                  >
+                                    Feature
+                                  </div>
+                                  <div
+                                    className='table-column'
+                                    style={{ width: metadataWidth[3] }}
+                                  >
+                                    Remove
+                                  </div>
+                                </Row>
+                              }
+                              renderRow={(metadata) => (
+                                <Row
+                                  space
+                                  className='list-item clickable cursor-pointer'
+                                  key={metadata.id}
+                                  onClick={() => {
+                                    this.editMetadata(
+                                      metadata.id,
+                                      metadata.content_type_fields,
+                                    )
+                                  }}
+                                >
+                                  <Flex className='table-column px-3'>
+                                    <div className='font-weight-medium mb-1'>
+                                      {metadata.name}
+                                    </div>
+                                  </Flex>
+                                  <div
+                                    className='table-column'
+                                    style={{
+                                      alignItems: 'center',
+                                      display: 'flex',
+                                      width: '185px',
+                                    }}
+                                  >
+                                    <div className='table-column'>
+                                      {metadata.content_type_fields.find(
```

Comment:
Hard to read what the following functions are doing, perhaps separate these into variables above the markup ?